src/rprblender/nodes/rpr_nodes.py

```diff
@@ -1513,3 +1513,95 @@ def export_hybrid(self):
+        self.inputs.new('NodeSocketColor', "Color").default_value = (0.8, 0.8, 0.8, 1.0)    # Corresponds to Cycles diffuse
+        self.inputs.new('NodeSocketFloatFactor', "Roughness").default_value = 1.0
+        self.inputs.new('NodeSocketVector', "Normal").hide_value = True
+
+        inp = self.inputs.new('NodeSocketColor', "Shadow Color")
+        inp.default_value = (0.0, 0.0, 0.0, 1.0)    # Corresponds to Cycles diffuse
+        inp.enabled = False
+        inp = self.inputs.new('NodeSocketFloatFactor', "Mid Level")
+        inp.default_value = 0.5
+        inp.enabled = False
+        inp = self.inputs.new('NodeSocketFloatFactor', "Mid Level mix")
+        inp.default_value = 0.05
+        inp.enabled = False
+        inp = self.inputs.new('NodeSocketColor', "Mid Color")
+        inp.default_value = (0.4, 0.4, 0.4, 1.0)    # Corresponds to Cycles diffuse
+        inp.enabled = False
+        inp = self.inputs.new('NodeSocketFloatFactor', "Highlight Level")
+        inp.default_value = 0.8
+        inp.enabled = False
+        inp = self.inputs.new('NodeSocketFloatFactor', "Highlight Level mix")
+        inp.default_value = 0.05
+        inp.enabled = False
+        inp = self.inputs.new('NodeSocketColor', "Highlight Color")
+        inp.default_value = (0.8, 0.8, 0.8, 1.0)    # Corresponds to Cycles diffuse
+        inp.enabled = False
+
+        # adding output socket
+        self.outputs.new('NodeSocketShader', "Shader")
+
+    def draw_buttons(self, context, layout):
+        col = layout.column()
+
+        col.prop(self, 'show_advanced')
+        if self.show_advanced:
+            col.prop(self, 'show_mix_levels')
+
+    class Exporter(RuleNodeParser):
+        def export(self):
+            if self.node.show_advanced:
+                # build the toon ramp node
+                interpolation_mode = pyrpr.INTERPOLATION_MODE_LINEAR if self.node.show_mix_levels \
+                    else pyrpr.INTERPOLATION_MODE_NONE
+                ramp = self.create_node(pyrpr.MATERIAL_NODE_TOON_RAMP, {
+                    pyrpr.MATERIAL_INPUT_SHADOW: self.get_input_value('Shadow Color'),
+                    pyrpr.MATERIAL_INPUT_MID: self.get_input_value('Mid Color'),
+                    pyrpr.MATERIAL_INPUT_HIGHLIGHT: self.get_input_value('Highlight Color'),
+                    pyrpr.MATERIAL_INPUT_POSITION1: self.get_input_value('Mid Level'),
+                    pyrpr.MATERIAL_INPUT_POSITION2: self.get_input_value('Highlight Level'),
+                    pyrpr.MATERIAL_INPUT_RANGE1: self.get_input_value('Mid Level Mix'),
+                    pyrpr.MATERIAL_INPUT_RANGE2: self.get_input_value('Highlight Level Mix'),
+                    pyrpr.MATERIAL_INPUT_INTERPOLATION: interpolation_mode,
+                })
+
+                return self.create_node(pyrpr.MATERIAL_NODE_TOON_CLOSURE, {
+                    pyrpr.MATERIAL_INPUT_COLOR: (1.0, 1.0, 1.0, 1.0),
+                    pyrpr.MATERIAL_INPUT_NORMAL: self.get_input_link('Normal'),
```

Comment:
The None value(unlinked socket) causes error here. Something like that would be better:

            normal = self.get_input_link('Normal')
            if self.node.show_advanced:
            ...
                node = self.create_node(pyrpr.MATERIAL_NODE_TOON_CLOSURE, {
                    pyrpr.MATERIAL_INPUT_COLOR: (1.0, 1.0, 1.0, 1.0),
                    pyrpr.MATERIAL_INPUT_ROUGHNESS: self.get_input_value('Roughness'),
                    pyrpr.MATERIAL_INPUT_DIFFUSE_RAMP: ramp
                })

            else:
                node = self.create_node(pyrpr.MATERIAL_NODE_TOON_CLOSURE, {
                    pyrpr.MATERIAL_INPUT_COLOR: self.get_input_value('Color'),
                    pyrpr.MATERIAL_INPUT_ROUGHNESS: self.get_input_value('Roughness')
                })

            if normal:
                node.set_input(pyrpr.MATERIAL_INPUT_NORMAL, normal)

            return node